infra/gravity/ops_center.go

```diff
@@ -0,0 +1,159 @@
+package gravity
+
+import (
+	"bufio"
+	"bytes"
+	"errors"
+	"fmt"
+	"io"
+	"os/exec"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	errClusterNotFound = errors.New("cluster not found")
+)
+
+// getTeleClusterStatus will attempt to get the cluster status from an ops center
+// for now, it's just looking for the specific line of output from tele get clusters
+// but could parse the entire yaml structure if imported from gravity
+func getTeleClusterStatus(clusterName string) (string, error) {
+	out, err := exec.Command("tele", "get", "clusters", clusterName, "--format", "yaml").Output()
+	if err != nil {
+		logrus.WithError(err).Error("unable to parse tele get clusters: ", string(out))
+		return "", trace.WrapWithMessage(err, string(out))
+	}
+
+	res, err := parseClusterStatus(clusterName, bytes.NewReader(out))
+	if err != nil {
+		logrus.WithError(err).Error("unable to parse tele get clusters: ", string(out))
+		return "", trace.WrapWithMessage(err, string(out))
+	}
+	return res, nil
+}
+
+// parseClusterStatus will look at the command output, and search for the string "status: abc" and return
+// the "abc" portion.
+func parseClusterStatus(clusterName string, rd io.Reader) (string, error) {
+	scanner := bufio.NewScanner(rd)
+	for scanner.Scan() {
+		trimmed := strings.TrimSpace(scanner.Text())
+		if strings.HasPrefix(trimmed, "status") {
+			// we found the status line
+			split := strings.Split(trimmed, ": ")
+			if len(split) < 2 {
+				return "", trace.BadParameter("invalid status line: %v", trimmed)
+			}
+			return split[1], nil
+		}
+
+		if strings.Contains(trimmed, fmt.Sprintf("cluster %v not found", clusterName)) {
+			return "", trace.Wrap(errClusterNotFound)
+		}
+	}
+
+	return "", trace.BadParameter("invalid input")
+}
+
+// generateClusterConfig will generate a cluster configuration for the ops center based
+// on the built in template
+func generateClusterConfig(cfg ProvisionerConfig, clusterName string) (string, error) {
+	template, err := template.New("cluster").Parse(`
```

Comment:
I would still move this to init block (make it a package global) - so that the parse errors can be caught as early as possible:
```go
var clusterConfigTemplate = template.Must(template.New("cluster").Parse(`...`))
```